lib/hammer_cli/output/definition.rb

```diff
@@ -12,10 +12,117 @@ def append(fields)
       @fields += fields
     end
 
+    def delete_fields!(by = {})
+      return clear if by.empty?
+      return deep_delete!(@fields, by) if by[:deep_delete]
+      ids, labels = adjust_options(by)
+      @fields.delete_if do |f|
+        label = f.label
+        label = f.label.downcase unless label.nil?
+        ids.include?(f.id) || labels.include?(label)
+      end
+    end
+
+    def select_fields!(by = {})
+      return self if by.empty?
+      return @fields = deep_select(@fields, by) if by[:deep_select]
+      ids, labels = adjust_options(by)
+      @fields.keep_if do |f|
+        label = f.label
+        label = f.label.downcase unless label.nil?
+        ids.include?(f.id) || labels.include?(label)
+      end
+    end
+
+    def replace_field(field_id, fields = nil, &block)
+      i = @fields.find_index do |f|
+        f.id == field_id
+      end
+      return if i.nil?
+      definition = build_definition(fields, &block)
+      multiple_replace(i, definition.fields)
+    end
+
+    def at(path = [])
+      return self if path.empty?
+      field = expand_path(@fields, path.dup)
+      return nil if field.nil? || !field.respond_to?(:output_definition)
+      field.output_definition
+    end
+
+    def clear
+      @fields = []
+    end
+
     def empty?
       @fields.empty?
     end
 
+    private
+
+    def build_definition(fields = nil, &block)
```

Comment:
We're using similar code at 2 places already. I think we could refactor it into a definition's public method:
```ruby
def build(dsl = Dsl.new, &block)
  dsl.build(&block) if block_given?
  append(dsl.fields)
end
```
and also if we changed append to:
```ruby
def append(fields)
  unless fields.nil?
    fields = [fields] unless fields.is_a? Array
    @fields += fields
  end
end
```
than it would be much easier for use at all the places and also decoupled from the dsl definition.

This method could then be just:
```ruby
definition = self.class.new
definition.append(fields)
definition.build(&block)
definition
```